activesupport/lib/active_support/hash_with_indifferent_access.rb

```diff
@@ -363,6 +363,14 @@ def to_hash
     end
 
     private
+      if Symbol.method_defined?(:name)
+        using Module.new {
+          refine Symbol do
```

Comment:
Why use a refinement? it could be implemented in a much simpler way without any overhead: https://github.com/rails/rails/pull/40201/files